e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/report-reconciliation/bank-utilisation-report-page.spec.js

```diff
@@ -0,0 +1,118 @@
+import {
+  CURRENCY,
+  FeeRecordEntityMockBuilder,
+  REPORT_NOT_RECEIVED,
+  RECONCILIATION_IN_PROGRESS,
+  UtilisationReportEntityMockBuilder,
+  toIsoMonthStamp,
+  getPreviousReportPeriodForBankScheduleByMonth,
+  FEE_RECORD_STATUS,
+  getFormattedReportPeriodWithLongMonth,
+} from '@ukef/dtfs2-common';
+import pages from '../../../pages';
+import USERS from '../../../../fixtures/users';
+import { NODE_TASKS } from '../../../../../../e2e-fixtures';
+
+const { utilisationReportPage } = pages;
+
+context('Bank utilisation report page', () => {
+  const SUBMISSION_MONTH = toIsoMonthStamp(new Date());
+  const BANK_ID = '961';
+  const REPORT_ID = 1;
+  const FEE_RECORD_ID_ONE = '11';
+  const FACILITY_ID_ONE = '11111111';
+  let bankName;
+  let reportPeriodString;
+
+  const aTfmFacilityFacilitySnapshot = () => ({
+    coverStartDate: new Date().getTime(),
+    coverEndDate: new Date().getTime(),
+    interestPercentage: 5,
+    dayCountBasis: 365,
+    coverPercentage: 80,
+    value: 100000,
+  });
+
+  const TFM_FACILITIES = [
+    {
+      facilitySnapshot: { ...aTfmFacilityFacilitySnapshot(), ukefFacilityId: FACILITY_ID_ONE },
+    },
+  ];
+
+  before(() => {
+    const visibleBanks = [];
+
+    cy.task(NODE_TASKS.DELETE_ALL_TFM_FACILITIES_FROM_DB);
+    cy.task(NODE_TASKS.INSERT_TFM_FACILITIES_INTO_DB, TFM_FACILITIES);
+
+    cy.task(NODE_TASKS.GET_ALL_BANKS).then((getAllBanksResult) => {
+      getAllBanksResult
+        .filter((bank) => bank.isVisibleInTfmUtilisationReports)
+        .forEach((visibleBank) => {
+          visibleBanks.push(visibleBank);
+        });
+    });
+
+    cy.task(NODE_TASKS.DELETE_ALL_FROM_SQL_DB);
+
+    cy.wrap(visibleBanks).each((bank) => {
+      const reportPeriod = getPreviousReportPeriodForBankScheduleByMonth(bank.utilisationReportPeriodSchedule, SUBMISSION_MONTH);
+
+      reportPeriodString = getFormattedReportPeriodWithLongMonth(reportPeriod);
+
+      if (bank.id === BANK_ID) {
+        bankName = bank.name;
+
+        const reportToReconcile = UtilisationReportEntityMockBuilder.forStatus(RECONCILIATION_IN_PROGRESS)
+          .withId(REPORT_ID)
+          .withBankId(BANK_ID)
+          .withReportPeriod(reportPeriod)
+          .build();
+        cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [reportToReconcile]);
+
+        const feeRecordOne = FeeRecordEntityMockBuilder.forReport(reportToReconcile)
+          .withId(FEE_RECORD_ID_ONE)
+          .withFacilityId(FACILITY_ID_ONE)
+          .withExporter('Exporter 1')
+          .withPaymentCurrency(CURRENCY.GBP)
+          .withFeesPaidToUkefForThePeriod(100)
+          .withFeesPaidToUkefForThePeriodCurrency(CURRENCY.JPY)
+          .withPaymentExchangeRate(2)
+          .withStatus(FEE_RECORD_STATUS.MATCH)
```

Comment:
I think we can delete all these lines since they aren't linking to any other set up or assertions so we can just use the default values the builder will populate which will reduce overall lines of code

Reply:
removed